### PR TITLE
upgrade to angular 9

### PR DESCRIPTION
**Pull request by RafaelLiendo**
Project rewritten using AngularCLI and following [the documentation guides on creating libraries](https://angular.io/guide/creating-libraries)
Everything still works the same, and tests ar passing.

### CODE REVIEW

**Comment by fals** [comment]
@RafaelLiendo thank you for the contribution. Could you please bump the package.json version to v9?

File: package.json [hunk 1]
```diff
@@ -1,68 +1,47 @@
 {
-    "name": "ng2-brpipes",
```
Comment (fals) [inline package.json:2]: this section cannot be removed otherwise I can't publish the package

**Comment by fals** [comment]
@RafaelLiendo you basically removed all the needed information from package.json to deploy the package on npm. Until it's fixed I can't merge it. Also I'm afraid I will have to change something else in the CI/CD pipeline because I wasn't using the CLI to deploy this package.

**Comment by RafaelLiendo** [comment]
Hi @fals, Sorry for the delay. Following the [documentation](https://angular.io/guide/creating-libraries), wich uses [ng-packagr](https://github.com/ng-packagr/ng-packagr) under the hood. All the package.json info related to the package (not the root project), goes inside 
[/projects/ng2-brpipes/package.json](https://github.com/RafaelLiendo/ng2-brpipes/blob/1d057debfc66c3606db19f0cd83fd00199472883/projects/ng2-brpipes/package.json#L1-L28).
Also, when you run
```bash
npm run build -- --prod
```
it generates a dist/ng2-brpipes folder that also contains a package-json
image.
that looks like this:
```json
{
  "name": "ng2-brpipes",
  "version": "9.0.0",
  "description": "Angular2 pipes for BR types",
  "author": "Filipe Augusto Lima de Souza",
  "repository": {
    "type": "git",
    "url": "https://github.com/fals/ng2-brpipes"
  },
  "license": "MIT",
  "keywords": [
    "ng2",
    "ng",
    "angular",
    "angular2",
    "angularjs",
    "pipes",
    "filters",
    "library"
  ],
  "peerDependencies": {
    "@angular/common": ">= 2",
    "@angular/core": ">= 2",
    "tslib": "^1.10.0"
  },
  "optionalDependencies": {
    "tslib": "^1.10.0"
  },
  "main": "bundles/ng2-brpipes.umd.js",
  "module": "fesm5/ng2-brpipes.js",
  "es2015": "fesm2015/ng2-brpipes.js",
  "esm5": "esm5/ng2-brpipes.js",
  "esm2015": "esm2015/ng2-brpipes.js",
  "fesm5": "fesm5/ng2-brpipes.js",
  "fesm2015": "fesm2015/ng2-brpipes.js",
  "typings": "ng2-brpipes.d.ts",
  "metadata": "ng2-brpipes.metadata.json",
  "sideEffects": false
}
```
I also updated the .circleci script to **cd** into the dist/ng2-brpipes, and then npm publish, so the online package.json that goes to npm repo is the dist one.
```yml
steps:
      - checkout
      - run:
            name: Publish to NPM
            command: |
              npm install
              npm set //registry.npmjs.org/:_authToken=$NPM_TOKEN
              npm run build -- --prod
              cd dist/ng2-brpipes
              npm publish
```

🙂 tell me if you need anything

**Comment by RafaelLiendo** [comment]
Hi again, I created a [repo](https://github.com/RafaelLiendo/ng2-brpipes-dist) with just the /dist/ng2-brpipes folder. So you could test it by adding it to your package.json
```json
dependencies: {
  "ng2-brpipes": "git+https://git@github.com//RafaelLiendo/ng2-brpipes-dist",
````

**Comment by RafaelLiendo** [comment]
Me again. I just installed it in a angular 6 project and it worked like a charm \o/
I'm gonna find a angular 2 project on github to see how far it goes

**Comment by RafaelLiendo** [comment]
YEESSS!!! It worked on angular 2, haha!

FYI if you ever need to install previous angular versions, here is a good article about it:
https://medium.com/ng-gotchas/easily-create-the-legacy-angular-apps-v2-v4-v5-v6-ee4a22d7eb60

**Pull request merged**